### PR TITLE
Reorganize rest_server template with new layout and subfolder convention

### PR DESCRIPTION
**Pull request by kevya-google**
Addresses https://github.com/abcxyz/abc/issues/173, also reformats subdirectory contents in accordance to https://github.com/abcxyz/abc/issues/166.

Template structure:
```
$ tree -all
.
├── contents
│   ├── go.mod
│   ├── go.sum
│   ├── main.go
│   ├── main_test.go
│   └── README.md
├── deployments
│   ├── config
│   │   └── autopush-rest-server.env
│   ├── contents
│   │   └── Dockerfile
│   ├── README.md
│   ├── spec.yaml
│   └── workflows
│       ├── ci-rest-server.yaml
│       └── deploy-rest-server.yaml
├── README.md
└── spec.yaml

```
Rendered template structure:
```
$ tree -all
.
├── Dockerfile
├── .github
│   ├── config
│   │   └── autopush-rest-server.env
│   └── workflows
│       ├── ci-rest-server.yaml
│       └── deploy-rest-server.yaml
├── go.mod
├── go.sum
├── main.go
├── main_test.go
└── README.md
```

### CODE REVIEW

**Review by drevell — APPROVED** [review]
Thanks for updating this

File: t/rest_server/workflows/contents/workflows/deploy-rest-server.yaml [hunk 1]
```diff
@@ -41,6 +41,9 @@ jobs:
       id-token: 'write'
     name: 'Release Image'
     runs-on: 'ubuntu-latest'
+    defaults:
+      run:
+        working-directory: './code/rest_server'
```
Comment (kevya-google) [inline t/rest_server/workflows/contents/workflows/deploy-rest-server.yaml:46]: Note: this workflow expects the opinionated folder structure, not sure how to work around this if --dest is omitted.

**Comment by kevya-google** [comment]
Converting back to draft to fix Dockerfile reference from deployments folder in deploy.yaml and getting consensus with @gjonathanhong on folder structure.

**Review by drevell — DISMISSED** [review]

**Comment by kevya-google** [comment]
Pending: will try adding input validation.

**Review by gjonathanhong — APPROVED** [review]

**Review by drevell — APPROVED** [review]

**Pull request merged**